DOC Document image lazy-loading (#9990)
silverstripe/silverstripe-framework: docs/en/04_Changelogs/4.9.0.md

# 4.9.0 (Unreleased)

## Features and enhancements {#features-and-enhancements}

### Image lazy-loading set by default

> HTML `<img>` tags now have the `loading="lazy"` attribute added by default on image that are:

Reviewer: ```suggestion
HTML `<img>` tags now have the `loading="lazy"` attribute added by default on images that are:
```

Author: Updated

- Inserted via the WYSIWYG editor

> - Added to a silverstripe template as an image object e.g. `$MyImage.SetWidth(300)`

Reviewer: Is the `SetWidth(300)` part necessary here?

Author: Nope, have removed

> - Added to a silverstripe template as an image object that redners as HTML e.g. `<div>$MyImage</div>`

Reviewer: ```suggestion
- Added to a SilverStripe template as an image object that renders as HTML, e.g. `<div>$MyImage</div>`
```

Author: Updated

> Other `<img>` tags such as those hardcoded in silverstripe templates and HTML returned in API responses remain unchanged.

Reviewer: ```suggestion
Other `<img>` tags such as those hardcoded in SilverStripe CMS templates and HTML returned in API responses remain unchanged.
```

Author: Updated

Individual images can have lazy-loading disabled via the following methods:

> - WYSIWYG images - Click on the image, then click the "Edit image" icon.  Modify the value in form control in the image placement form

Reviewer: ```suggestion
- WYSIWYG images - Click on the image, then click the "Edit image" icon.  Modify the lazy-loading option in the image placement form
```

Author: We haven't yet determined what the text for this will be, it may be 'lazy-loading', it may be 'eager-loading', it may be something else.  We can update this once the UX component has been desigend.  I'll add a point to the [UX card](https://github.com/silverstripe/silverstripe-asset-admin/issues/1214)

> - Image objects in template - suffix `.LazyLoad(false)` in the silverstripe template (to be confirmed)

Reviewer: ```suggestion
- Image objects in templates - suffix with `.LazyLoad(false)` in the SilverStripe CMS template (*to be confirmed*)
```

Author: Updated

> - Image objects in templates - suffix with `.LazyLoad(false)` in the SilverStripe CMS template (*to be confirmed*)

Reviewer: ```suggestion
- Image objects in templates - suffix with `.LazyLoad(0)` in the SilverStripe CMS template
```

SilverStripe template syntax does not handle booleans yet, but this is something we could consider cherry-picking. https://github.com/silverstripe/silverstripe-assets/pull/457#discussion_r662835648

Author: looks like there's still some discussion here, it would be preferable to do LazyLoad(false), even if that's actually just passing a string "false".  I'll just leave as is for now, we can easily update these docs later
